src/shr_transport.h4

```diff
@@ -591,4 +568,51 @@ SHMEM_DEFINE_FOR_AMO(SHMEM_DEF_SUM_OP)
 #endif
 }
 
+
+static inline void
+shmem_shr_transport_put_signal(shmem_ctx_t ctx, void *target,
+                               const void *source, size_t len,
+                               uint64_t *sig_addr, uint64_t signal, int sig_op, int pe)
+{
+#if USE_MEMCPY
+    memcpy(target, source, len);
+    if (sig_op == SHMEM_SIGNAL_ADD) *sig_addr += signal;
+    else *sig_addr = signal;
+#elif USE_XPMEM
+    shmem_transport_xpmem_put(target, source, len, pe,
+                              shmem_internal_get_shr_rank(pe));
+    shmem_internal_membar_acq_rel(); /* Memory fence to ensure target PE observes
+                                        stores in the correct order */
+    if (ONLY_XPMEM_TRANSPORT) {
```

Comment:
This should be replaced by `#if USE_SHR_ATOMICS` but other than change this looks looks ok.